### PR TITLE
Support tsdb version from schema config in head manager

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

This is PR 2/3 to move the TSDB schema version to the period config. This PR sets the correct version in the index file when cutting the index from the head manager (aka ingesters). Future work will be to also respect this versioning in the compactor.

**Which issue(s) this PR fixes**:
Re #9565

**Special notes for your reviewer**:

The full functionality of supporting TSDB index versions in the schema/period config is enabled via 3 PRs. Each PR can be reviewed and merged independently.

[1/3] https://github.com/grafana/loki/pull/9566
[2/3] https://github.com/grafana/loki/pull/9590
[3/3] https://github.com/grafana/loki/pull/9633

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

File: operator/website/node_modules
Comment (periklis) [inline operator/website/node_modules:1]: This file is probably accidentally in this PR.

Comment (trevorwhitney) [reply operator/website/node_modules:1]: My editor's language server was complaining about this symlink missing, so I removed it. Is there a better way to fix that? I can remove this change from this PR, but curious going forward how best to handle this?

**Comment by trevorwhitney** [comment]
This behavior was intended to enable faster rollouts of new TSDB indexes for quicker iteration. As we are not currently iterating on the index as we thought we may have been, putting this into draft for another day.

**Pull request closed without merging**